src/libraries/System.Net.Mail/src/System/Net/Mime/QEncodedStream.cs

```diff
@@ -233,30 +239,28 @@ public override void Write(byte[] buffer, int offset, int count)
             }
         }
 
-        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default)
         {
-            ValidateBufferArguments(buffer, offset, count);
-            return WriteAsyncCore(buffer, offset, count, cancellationToken);
-
-            async Task WriteAsyncCore(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+            int written = 0;
+            while (true)
```

Comment:
This loop isn't really a loop. It'll always return in the first iteration.

Isn't `FlushAndContinueAsync` potentially a stack overflow issue if things complete synchronously? It might be easier to keep this method async as it was.

Reply:
Reverted